### DISCUSSION TITLE
SOLID for Raspberry Pi 4でVLINKの機能は使えますか？

### DISCUSSION THREAD

**Discussion by asano-tk** [KyotoMicrocomputer/solid-rapi4#3] [Q&A] [answered]
SOLID for Raspberry Pi 4とSolid-IDEを使って開発を行っております。

https://solid.kmckk.com/SOLID/doc/latest/os/filesystem.html
```
int fd;
int result = SOLID_FS_Open(&fd, "\\VLINK\\textfile.txt", O_CREAT | O_RDWR | O_TRUNC);
if (result == SOLID_ERR_OK)
{
    size_t bytes_written;
    result = SOLID_FS_Write(fd, "hello", 5, &bytes_written);
    SOLID_FS_Close(fd);
}
```
上記ページにある上記のコードを実行しても、ファイルが生成されません。
ブレークポイントを設定してresultの値を確認した所、-1000(SOLID_ERR_NOTFOUND)となっていました。

また、
SOLID_DBG_IsInDebugger()の戻り値は「1:接続されている(デバッグ中)」
SOLID_VLINK_IsAvailable()の戻り値は「0:使用不可」
となっています。

SOLID for Raspberry Pi 4でVLINKの機能は使えないのでしょうか？
または、何か設定が必要なのでしょうか？

■開発環境
![image](https://github.com/KyotoMicrocomputer/solid-rapi4/assets/93647368/de8ccd24-74f4-4e09-b179-76b0ddaed878)

**Comment by kawadakk** [comment, marked as answer]
SOLID for Raspberry Pi 4環境ではVLINKは対応しておりません。

Raspberry Pi 4上でSOLID-OSからLinuxのファイルシステムを[アクセスできる](https://github.com/KyotoMicrocomputer/solid-rapi4/blob/main/doc/system-rtos.md#os%E9%96%93%E9%80%9A%E4%BF%A1-oscom-rpc)ようにしているので、Linuxを介してファイル共有をする方法をご検討ください。

具体的には、[SSH経由でアップロード](https://github.com/KyotoMicrocomputer/solid-rapi4/blob/main/doc/system-linux.md#secure-copy)、または`mount.cifs` プログラムで[Windows共有フォルダをLinuxファイルシステムにマウント](https://www.raspberrypi.com/documentation/computers/remote-access.html#mount-the-folder-on-the-raspberry-pi)し、SOLIDファイルシステムAPIから `\OSCOM_FS\home\pi\xxx` (`/home/pi/xxx` に配置した場合) のようなパスを指定してアクセスしてください。

**Reply by asano-tk** [reply]
ご回答ありがとうございます。
ご提案頂いた内容で改めて検討してみます。
